src/literacy.rs

```diff
@@ -0,0 +1,148 @@
+#[cfg(all(feature = "std", feature = "use-core2"))]
+compile_error!("feature \"std\" and \"use-core2\" cannot be enabled together.");
+
+#[derive(Debug)]
+pub enum Error {
+    #[cfg(feature = "std")]
+    Std(::std::io::Error),
```

Comment:
Feature-gated enums are really bad because cargo uses the union set of all features in the dependency tree (e.g. if there are two deps with a rust-bitcoin dep and one rust-bitcoin has use-core and another std cargo will compile it once with both features. That means a non-breaking change (just changing the feature set of rust-bitcoin without exposing it) in one of the deps could break  exhaustive matches.

Would `Box<dyn SomeNonStdErrorTrait>` work here?

Reply:
I thought of `Box<dyn SomeNonStdErrorTrait>` for the `Other` variant, but I am not sure what `SomeNonStdErrorTrait` could be, `Debug`? Or something created ad-hoc?

Reply:
It could be a new trait defined as `trait OurErrorTrait: Dsiplay + Debug + … {}`, essentially just a hack to make multiple trait bounds object safe.

Reply:
I think we can't avoid the feature gate on the error variant on 1.29, because you can't use `Box` taken from alloc on 1.29

Reply:
With 1.29, you must turn on std and must not turn on use-core2 anyway, so there seems to be no issue.  Box only needs to be imported from `alloc` conditional on `not(feature = "std"))`, which will be skipped by 1.29.

Reply:
Yeah, right, I had to update the CI for that (skipping no_std for 1.29) 4727b6364850764ce09ba210d0dae27d952d6b5a

Reply:
I'm a bit confused as to how CI was passing before, given that 1.29 can't compile core2.